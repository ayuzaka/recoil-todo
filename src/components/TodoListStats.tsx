import { VFC } from 'react'
import { useStatsTodoList } from '../hooks/useStatesTodoList'

type Props = {
  totalNum: number
  totalCompletedNum: number
  totalUncompletedNum: number
  formattedPercentCompleted: number
}

const Component: VFC<Props> = ({ totalNum, totalCompletedNum, totalUncompletedNum, formattedPercentCompleted }) => (
  <ul>
    <li data-testid="totalItem">Total Item: {totalNum}</li>
    <li data-testid="completedItem">Items completed: {totalCompletedNum}</li>
    <li data-testid="uncompletedItem">Items not completed: {totalUncompletedNum}</li>
    <li data-testid="percentCompleted">Percent completed: {formattedPercentCompleted}</li>
  </ul>
)

const Container: VFC = () => {
  const { totalNum, totalCompletedNum, totalUncompletedNum, percentCompleted } = useStatsTodoList()
  const formattedPercentCompleted = Math.round(percentCompleted)

  return <Component {...{ totalNum, totalCompletedNum, totalUncompletedNum, formattedPercentCompleted }} />
}

export const TodoListStats = Container
