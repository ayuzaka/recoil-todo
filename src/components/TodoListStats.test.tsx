import { RecoilRoot } from 'recoil'
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import { TodoListStats } from './TodoListStats'

describe('初期状態', () => {
  test('render', () => {
    const { asFragment } = render(
      <RecoilRoot>
        <TodoListStats />
      </RecoilRoot>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test('初期状態はすべてが0であること', () => {
    render(
      <RecoilRoot>
        <TodoListStats />
      </RecoilRoot>
    )

    const totalItem = screen.getByTestId('totalItem')
    expect(totalItem).toHaveTextContent('Total Item: 0')

    const completedItem = screen.getByTestId('completedItem')
    expect(completedItem).toHaveTextContent('Items completed: 0')

    const uncompletedItem = screen.getByTestId('uncompletedItem')
    expect(uncompletedItem).toHaveTextContent('Items not completed: 0')

    const percentCompleted = screen.getByTestId('percentCompleted')
    expect(percentCompleted).toHaveTextContent('Percent completed: 0')
  })

  test.todo('ステートの値が反映されること')
})
