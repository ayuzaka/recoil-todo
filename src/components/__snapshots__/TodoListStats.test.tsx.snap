// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`初期状態 render 1`] = `
<DocumentFragment>
  <ul>
    <li
      data-testid="totalItem"
    >
      Total Item: 0
    </li>
    <li
      data-testid="completedItem"
    >
      Items completed: 0
    </li>
    <li
      data-testid="uncompletedItem"
    >
      Items not completed: 0
    </li>
    <li
      data-testid="percentCompleted"
    >
      Percent completed: 0
    </li>
  </ul>
</DocumentFragment>
`;
